drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -735,7 +735,7 @@ void RasterizerCanvasGLES3::_render_items(RID p_to_render_target, int p_item_cou
 
 				} break;
 				case GLES3::CanvasShaderData::BLEND_MODE_SUB: {
-					glBlendEquation(GL_FUNC_REVERSE_SUBTRACT);
+					glBlendEquation(GL_FUNC_SUBTRACT);
```

Comment:
I'm not sure about this. In Godot ``BLEND_MODE_SUB`` means that the color from the pixel shader is subtracted from the color behind it. That corresponds to ``GL_FUNC_REVERSE_SUBTRACT``. ``GL_FUNC_SUBTRACT`` subtracts the background color from the pixel shader color and returns that value.

I.e. if you want to write a white pixel to a grey background with ``GL_FUNC_REVERSE_SUBTRACT`` you get ``(0.5, 0.5, 0.5) - (1, 1, 1,) = (0, 0, 0)`` while with ``GL_FUNC_SUBTRACT`` you get ``(1, 1, 1) - (0.5, 0.5, 0.5) = (0.5, 0.5, 0.5)``

Reply:
Ah, I see that the OpenGL renderer has always used `GL_FUNC_REVERSE_SUBTRACT` for `BLEND_MODE_SUB`. Thanks for pointing this out! I will remove this from the PR.

The reason that I initially made this change was to match the behaviour of the other two renderers in Godot 4 which use `BLEND_OP_SUBTRACT` instead of `BLEND_OP_REVERSE_SUBTRACT`:

https://github.com/godotengine/godot/blob/b7032b5ecf8ac247e01acd5e54c9028c0dfe52df/servers/rendering/renderer_rd/forward_mobile/scene_shader_forward_mobile.cpp#L221-L222

https://github.com/godotengine/godot/blob/b7032b5ecf8ac247e01acd5e54c9028c0dfe52df/servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp#L210-L211

https://github.com/godotengine/godot/blob/b7032b5ecf8ac247e01acd5e54c9028c0dfe52df/servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp#L2124-L2125

I will open a different issue sometime later regarding the inconsistency of `BLEND_MODE_SUB` when using different renderers in Godot 4.

Reply:
Ah, good catch. So it seems the RD backends are reversed. That actually explains some things